src/Components/Actions/AddressAutocompleteAction/AddressAutocompleteAction.tsx

```diff
@@ -0,0 +1,580 @@
+import { motion } from 'framer-motion'
+import { StoreContext, Store } from '../../KeyValueStore'
+import { Tooltip } from '../../Tooltip'
+import { Card as BaseCard, Input, Spacer } from '../Common'
+import styled from '@emotion/styled'
+import { ApiContext } from '../../API/ApiContext'
+import { ApiComponent } from '../../API/apiComponent'
+import { useCombobox } from 'downshift'
+import {
+  AddressSuggestion,
+  CompleteAddress,
+  isCompleteAddress,
+} from '../../API/addressAutocomplete'
+import { colorsV3, fonts } from '@hedviginsurance/brand'
+import useDebounce from './useDebounce'
+import { ContinueButton } from '../../ContinueButton'
+import Modal from './Modal'
+import useAddressSearch from './useAddressSearch'
+import { isMatchingStreetName } from './utils'
+import { Cross } from '../../Icons/Cross'
+
+const ADDRESS_NOT_FOUND = 'ADDRESS_NOT_FOUND'
+
+const STORE_KEY = {
+  ID: 'bbrId',
+  ADDRESS: 'fullAddress',
+  STREET: 'street',
+  STREET_NAME: 'streetName',
+  STREET_NUMBER: 'streetNumber',
+  ZIP_CODE: 'zipCode',
+  CITY: 'city',
+  FLOOR: 'floor',
+  APARTMENT: 'apartment',
+  ADDRESS_SEARCH_TERM: 'addressSearchTerm',
+}
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: flex-start;
+`
+
+const Card = styled(BaseCard)`
+  width: 100%;
+
+  padding-bottom: 24px;
+
+  @media (max-width: 600px) {
+    padding-bottom: 16px;
+  }
+`
+
+const FakeInput = styled(Input)`
+  width: 100%;
+  text-align: left;
+  margin-left: 0;
+  margin-right: 0;
+  padding: 0 16px;
+  line-height: 1;
+  font-size: 48px;
+
+  @media (min-width: 600px) {
+    margin-top: 20px;
+    padding: 0 32px;
+  }
+`
+
+const ModalHeader = styled.header`
+  box-sizing: border-box;
+  width: 100%;
+  flex-shrink: 0;
+
+  padding: 0 16px 16px;
+  border-bottom: 1px solid ${colorsV3.gray300};
+`
+
+const ModalHeaderRow = styled.div<{ align: 'center' | 'flex-start' }>`
+  height: 56px;
+  display: flex;
+  align-items: ${(props) => props.align};
+  justify-content: flex-end;
+`
+
+const ModalHeaderLabel = styled.label`
+  position: absolute;
+  left: 0;
+  right: 0;
+  z-index: 1;
+
+  color: ${colorsV3.black};
+  font-family: ${fonts.FAVORIT};
+  text-align: center;
+  font-size: 16px;
+`
+
+const ModalHeaderButton = styled.button`
+  border: 0;
+  background: transparent;
+  appearance: none;
+  outline: 0;
+  position: relative;
+  z-index: 2;
+
+  color: ${colorsV3.black};
+  font-family: ${fonts.FAVORIT};
+  font-size: 16px;
+
+  &:focus {
+    border-radius: 4px;
+    box-shadow: 0 0 0 3px ${colorsV3.purple300};
+  }
+`
+
+const ComboboxField = styled.div<{ focus: boolean }>`
+  box-sizing: border-box;
+  text-align: left;
+  width: 100%;
+  position: relative;
+
+  padding: 8px 16px;
+  border-radius: 8px;
+  border: 1px solid
+    ${(props) => (props.focus ? colorsV3.purple300 : colorsV3.gray500)};
+  ${(props) => props.focus && `box-shadow: 0 0 0 2px ${colorsV3.purple300}`}
+`
```

Comment:
This works perfectly fine, just sharing an alternative solution for focus state that's kind of neat IMO :) With `:focus-within` you can omit the `isInputFocus` state and write it like so. Not supported in IE though 

```suggestion
const ComboboxField = styled.div`
  box-sizing: border-box;
  text-align: left;
  width: 100%;
  position: relative;

  padding: 8px 16px;
  border-radius: 8px;
  border: 1px solid colorsV3.gray500;

   &:focus-within {
    border-color: ${colorsV3.purple500};
    box-shadow: 0 0 0 2px ${colorsV3.purple300}
   }  
`
```

I would assume Mange and Zak would like to have a darker tint of gray instead of purple for the focus state, but we can go over that later 👍

Reply:
Nice! I like that solution -- and at this point, non-critical CSS should not have to consider IE